### DISCUSSION TITLE
Rust usage inside React Native

### DISCUSSION THREAD

**Discussion by jsoneaday** [react-native-community/discussions-and-proposals#466] [Ideas]
I know everyone is busy with the effort to release the new architecture, but as a question looking to the future. Since Meta seems to be a company that uses Rust, I was wondering if longer term there are any plans to switch out the C++ under pinnings of React Native for Rust?

I have no issue with continued use of C++. I'm just trying to plan ahead in terms of language learning that's all.

**Comment by ShikaSD** [comment]
Hey, I think React Native will continue using C++ in the visible future. Some of us are very excited about Rust and we are thinking about possible ways we can integrate it at some point.

I am pretty sure, however, that the core will continue using C++, as moving it to Rust will mean rewriting it from scratch. We might be exposing some bindings for extending Fabric core elements, but it is not planned for now.

**Reply by jsoneaday** [reply]
Nice thanks. What about webassembly. Any use cases for that? May not fit since rn is ultimately using native code. But it might be nice to take code written to run on web, maybe AI/ML, and also run it on both ios and android.

**Reply by ShikaSD** [reply]
I guess you try to run it as long as the JS engine you are using supports WASM, but I don't think there's going to be an official support in the nearest future.

**Comment by ivanmoskalev** [comment]
FYI, Rust has issues with Apple Bitcode (since it uses a version of LLVM that is different from the version bundled with Xcode).

**Reply by RichardFevrier** [reply]
Apple as since deprecated bitcode, so not a point anymore 🙂

**Reply by ivanmoskalev** [reply]
Yes! Good riddance!
